app/agency/utils.py

```diff
@@ -30,21 +36,66 @@ def update_agency_active_status(agency_ein, is_active):
             Agencies,
             agency_ein
         )
-        create_object(
-            Events(
-                request_id=None,
-                user_guid=current_user.guid,
-                auth_user_type=current_user.auth_user_type,
-                type_=AGENCY_ACTIVATED,
-                new_value={"ein": agency_ein},
-                timestamp=datetime.utcnow()
+        if is_active == "true":
+            create_object(
+                Events(
+                    request_id=None,
+                    user_guid=current_user.guid,
+                    auth_user_type=current_user.auth_user_type,
+                    type_=AGENCY_ACTIVATED,
+                    previous_value={"ein": agency_ein, "is_active": "False"},
+                    new_value={"ein": agency_ein, "is_active": "True"},
+                    timestamp=datetime.utcnow()
+                )
             )
-        )
-        # create request documents
-        for request in agency.requests:
-            request.es_create()
-
-        return True
+            # create request documents
+            for request in agency.requests:
+                request.es_create()
+
+            return True
+        if is_active == "false":
+            create_object(
+                Events(
+                    request_id=None,
+                    user_guid=current_user.guid,
+                    auth_user_type=current_user.auth_user_type,
+                    type_=AGENCY_DEACTIVATED,
+                    previous_value={"ein": agency_ein, "is_active": "True"},
+                    new_value={"ein": agency_ein, "is_active": "False"},
+                    timestamp=datetime.utcnow()
+                )
+            )
+            # remove requests from index
+            for request in agency.requests:
+                request.es_delete()
+            # deactivate agency users
+            active_users = get_agency_active_users(agency_ein)
+            for user in active_users:
```

Comment:
We should use the relationship `agency.active_users` and just iterate `for user in agency.active_users:`